From 21a9ac3a7e2554f94b8ff8788bc4e82ec3fb58d3 Mon Sep 17 00:00:00 2001
From: Gergely Nagy <algernon@balabit.hu>
Date: Mon, 3 Sep 2012 11:41:36 +0200
Subject: [PATCH] afsocket: Fix compilation without spoof-source

AFInetDestDriver->spoof_source is only available when compiling with
spoof-source enabled, so guard it with an #if.

Reported and patched by Dave Reisner <dreisner@archlinux.org>.

Signed-off-by: Gergely Nagy <algernon@balabit.hu>
---
 modules/afsocket/afinet.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/modules/afsocket/afinet.c b/modules/afsocket/afinet.c
index 5ca9172..98ff8d8 100644
--- a/modules/afsocket/afinet.c
+++ b/modules/afsocket/afinet.c
@@ -539,11 +539,13 @@ afinet_dd_setup_socket(AFSocketDestDriver *s, gint fd)
 static gboolean
 afinet_dd_init(LogPipe *s)
 {
-  AFInetDestDriver *self = (AFInetDestDriver *) s;
+  AFInetDestDriver *self G_GNUC_UNUSED = (AFInetDestDriver *) s;
   gboolean success;
 
+#if ENABLE_SPOOF_SOURCE
   if (self->spoof_source)
     self->super.flags &= ~AFSOCKET_KEEP_ALIVE;
+#endif
 
   success = afsocket_dd_init(s);
 #if ENABLE_SPOOF_SOURCE
-- 
1.7.12

